Makefile

```
@@ -73,13 +73,15 @@ run: build-local-container kind-load deploy ## Build image and run operator in-c
 
 # Binary builds
 GO_BUILD := $(Q)go build
+GOOS := $(or $(GOOS), linux)
+GOARCH := $(or $(GOARCH), amd64)
 build: bin/plain-v0 bin/unpack
 
 bin/plain-v0:
-	CGO_ENABLED=0 go build -o $@ ./provisioner/plain-v0
+	CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) go build -o $@ ./provisioner/plain-v0
 
 bin/unpack:
-	CGO_ENABLED=0 go build -o $@ ./cmd/unpack/...
+	CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) go build -o $@ ./cmd/unpack/...
 
 build-container: ## Builds provisioner container image locally
```

I have an Apple Silicon Mac. I think what we want here is:
`make bin/*` - by default does not set GOOS or GOARCH. That way the local build works for the local os/arch.
`make build-container` - by default, builds a linux binary, but uses GOARCH from the local environment. The assumption would be that the developer has a docker daemon running for their host OS's architecture.

I was actually thinking _just_ this:
```suggestion
build-container: GOOS=linux
build-container: ## Builds provisioner container image locally
```

---

Sounds good, updated